src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1479,8 +1522,115 @@ public void M(int[] a)
+                //         _ = a is [..{ Length: -1 }];  // 9
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is [..{ Length: -1 }]").WithArguments("int[]").WithLocation(21, 13),
+                // (22,13): error CS8518: An expression of type 'int[]' can never match the provided pattern.
+                //         _ = a is [..{ Length: < -1 }];  // 10
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is [..{ Length: < -1 }]").WithArguments("int[]").WithLocation(22, 13),
+                // (23,13): error CS8518: An expression of type 'int[]' can never match the provided pattern.
+                //         _ = a is [..{ Length: <= -1 }]; // 11
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is [..{ Length: <= -1 }]").WithArguments("int[]").WithLocation(23, 13),
+                // (26,13): error CS8518: An expression of type 'int[]' can never match the provided pattern.
+                //         _ = a is [_, _, ..{ Length: int.MaxValue - 1 }]; // 12
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is [_, _, ..{ Length: int.MaxValue - 1 }]").WithArguments("int[]").WithLocation(26, 13),
+                // (28,13): error CS8518: An expression of type 'int[]' can never match the provided pattern.
+                //         _ = a is [_, _, ..{ Length: >= int.MaxValue - 1 }]; // 13
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is [_, _, ..{ Length: >= int.MaxValue - 1 }]").WithArguments("int[]").WithLocation(28, 13),
+                // (30,13): error CS8518: An expression of type 'int[]' can never match the provided pattern.
+                //         _ = a is [_, _, ..{ Length: > int.MaxValue - 1 }]; // 14
+                Diagnostic(ErrorCode.ERR_IsPatternImpossible, "a is [_, _, ..{ Length: > int.MaxValue - 1 }]").WithArguments("int[]").WithLocation(30, 13)
+            );
+        }
+
+        [Fact]
+        public void ImpossiblePattern_02()
+        {
+            var source = @"
+interface IIndexable
+{
+    int this[int i] { get; }
+}
+interface ICountableViaCount
+{
+    int Count { get; }
+}
+interface ICountableViaLength
+{
+    int Length { get; }
+}
+class X
+{
+    public static void Test1<T>(T t) where T : ICountableViaCount
+    {
+        _ = t is { Count: -1 }; // ok
+    }
+    public static void Test2<T>(T t) where T : ICountableViaLength
+    {
+        _ = t is { Length: -1 }; // ok
+    }
+    public static void Test3<T>(T t) where T : IIndexable, ICountableViaCount
+    {
+        _ = t is { Count: -1 }; // 1
+    }
+    public static void Test4<T>(T t) where T : IIndexable, ICountableViaLength
+    {
+        _ = t is { Length: -1 }; // 2
+    }
+    public static void Test5<T>(T t) where T : IIndexable, ICountableViaLength, ICountableViaCount
+    {
+        _ = t is { Length: -1 }; // 3
+        _ = t is { Count: -1 }; // ok
```

Comment:
I find this behavior odd (although it matches the spec we agreed). I'll add an LDM open issue to propose that both be errors. What do you think?

Reply:
While there's no real-world scenario, I'd agree that doing this for both makes things less confusing in case the second property is added at a later point.

Reply:
It looks like the spec already implies that both should be errors: "Length or Count properties are assumed to always return a non-negative value, if and only if the type is indexable."

Reply:
I didn't mean it that way because "or" implies "either". Should be changed to "and" and we're on the same page.
Will go ahead and remove the symbol check. Best to be confirmed in LDM anyways though.